contracts/core/Registry.sol

```diff
@@ -147,6 +147,12 @@ contract Registry is IRegistry, Native, AccessControl, Transfer, Initializable,
 
         // Assign roles for the profile members
         uint256 memberLength = _members.length;
+
+        // Only profile owner can add members
+        if (memberLength > 0 && _owner != msg.sender) {
```

Comment:
This ensures the msg.sender cannot add members on behalf of the owner and this action is restricted only to the profile owner